lib/galaxy/tools/imp_exp/__init__.py

```diff
@@ -197,6 +197,24 @@ def get_tag_str( tag, value ):
                             raise Exception( "Invalid dataset path: %s" % temp_dataset_file_name )
                         if datasets_usage_counts[ temp_dataset_file_name ] == 1:
                             shutil.move( temp_dataset_file_name, hda.file_name )
```

Comment:
This is pre-existing code but I assume it informed the code below. It should use the object store, e.g.:

```
self.app.object_store.update_from_file( hda.dataset, file_name=temp_dataset_file_name, create=True )
```